### ISSUE TITLE
App is killed after 10 seconds when uninstalling

### ISSUE THREAD

**Issue by goaaats** [clowd/Clowd.Squirrel#135]
Hey! 

It looks like Squirrel kills my app after 10 seconds when uninstalling, which isn't ideal as I'm trying to show my users a prompt that goes "Do you want to delete your settings?".

https://github.com/clowd/Clowd.Squirrel/blob/develop/src/Squirrel/UpdateManager.ApplyReleases.cs#LL101C45-L101C45

In the docs however, this isn't mentioned, instead it just tells me to "[Exit as soon as you're finished](https://github.com/clowd/Clowd.Squirrel/blob/c8f6505b2de90f3a63ec7f37d7d376150902d356/docs/using/custom-squirrel-events-non-cs.md#L37)" which is actually what I want it to do.

What's the actual intended behaviour? It would be nice to be able to do some cleanup on my side, without having to worry about being killed if I take more than 10 seconds/potentially leaving some stuff behind, even without the potentially blocking user interaction.

Thanks!

**Comment by caesay** [comment]
I agree, the documentation is a bit misleading here. "Exit as soon as you're finished" at the moment means "Exit as quickly as possible after completing some automated cleanup". All of the Squirrel hooks expect you to exit within a certain amount of seconds, and it's not meant for you to show any UI. 

I am in the middle of some architecture changes for v3 and will keep this in mind to see if this can be improved. We have also had people asking to show custom UI during installation, so I just need to have a think about what's the best way to make this possible.
